types.go

```diff
@@ -199,3 +200,71 @@ func IsHostname(target string) bool {
 
 	return len(r) > 0
 }
+
+// DetectAssetTypes detects the asset types from a identifier.
+func DetectAssetTypes(identifier string) ([]string, error) {
+	if IsAWSARN(identifier) {
```

Comment:
It is a bit weird that any valid AWS ARN is considered an AWS account. Should we create a function `IsAWSAccount` that adds extra validations?

Reply:
@manelmontilla @julianvilas @jesusfcr what do you think?

The current logic is exactly the same used by vulcan-api in [github.com/adevinta/vulcan-api/pkg/api/service.getTypesFromIdentifier](https://github.com/adevinta/vulcan-api/blob/496a7a472a648c975d14640c6bdd160ef0ea1767/pkg/api/service/assets.go#L680).

Reply:
I think we should restrict the arn to just accounts, as this accepts any arn.
arn:aws:sqs:eu-west-1:892351162211:HermesAlertsSQS

Reply:
Regarding to the overall pr I think it's great to have this unified in one place, I already missed this when working in `vulcan-local` so I decided to clone and adapt the code.

Reply:
I agree that is weird, I guess there was a reason for doing that way but I honestly I don't remember it. I also agree with @jesusfcr in the sense that we must restrict the account ARN's and nothing else (I'm giving by granted that we want to use the format of the root user of an account: `arn:aws:iam::XXXX:root` as the only valid format for an AWS account) . In any case, I would not include this modification in this PR.

Reply:
Please check 5f0316d.

Reply:
@manelmontilla sorry, I read your comment just after pushing the fix. Given that we are not introducing breaking changes, I think it makes sense keep the commit and have the change available for other packages.

I assumed the format `arn:aws:iam::XXXX:root` based on [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html#principal-accounts).

Of course, I'm fine with dropping the commit and add it in a different PR.

Reply:
No, okey as it is already done and as you said, it doesn't introduce breaking changes, I think it's worth keeping.